src/app/components/tasks/tasks.component.spec.ts

```diff
@@ -0,0 +1,141 @@
+// @angular
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+// components
+import { TasksComponent } from './tasks.component';
+import { TaskComponent } from '../task/task.component';
+
+// services
+import { TaskService } from '../../services/task.service';
+
+// models
+import { Task } from '../../models/task.model';
+
+// mocks
+import { TaskServiceMock } from '../../mocks/mock.task.service';
+
+describe('TasklistComponent', () => {
+
+  let component: TasksComponent;
+  let fixture: ComponentFixture<TasksComponent>;
+  let taskService: TaskService;
+  let task: Task;
+
+  const completedTask: Task = { name: 'task 1', isCompleted: true };
+  const activeTask: Task = { name: 'task 2', isCompleted: false };
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      declarations: [ TasksComponent, TaskComponent ],
+      providers: [
+        { provide: TaskService, useClass: TaskServiceMock }
+      ]
+    }).compileComponents();
+
+    taskService = TestBed.get(TaskService);
+    fixture = TestBed.createComponent(TasksComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('call removeCompletedTasks() => call taskService.removeTasks()', () => {
+    const spy = spyOn(taskService, 'removeTasks').and.stub();
+    component.removeCompletedTasks();
+    expect(spy).toHaveBeenCalled();
+  });
+
+  it('call toggleTask(task) => call taskService.toggleTask(task)', () => {
+    const spy = spyOn(taskService, 'toggleTask').and.stub();
+    component.toggleTask(task);
+    expect(spy).toHaveBeenCalledWith(task);
+  });
+
+  it('call updateTask(task) => call taskService.updateTask(task)', () => {
+    const spy = spyOn(taskService, 'updateTask').and.stub();
+    component.updateTask(task);
+    expect(spy).toHaveBeenCalledWith(task);
+  });
+
+  it('call removeTask(task) => call taskService.removeTask(task)', () => {
+    const spy = spyOn(taskService, 'removeTask').and.stub();
+    component.removeTask(task);
+    expect(spy).toHaveBeenCalledWith(task);
+  });
+
+  it('call toggleAllTasks() => call taskService.toggleAllTasks(isChecked)', () => {
+    const spy = spyOn(taskService, 'toggleAllTasks').and.stub();
+    const isChecked = component.isAllCompleted();
+    component.toggleAllTasks();
+    expect(spy).toHaveBeenCalledWith(!isChecked);
+  });
+
+  it('call showActiveTasks() => set filter value equal to "active"', () => {
+    component.showActiveTasks();
+    const filter = component.filter;
```

Comment:
`const filter` not necessary, you can use component.filter as is.

Reply:
done

Reply:
done

Reply:
done